app/code/Magento/Customer/Test/Unit/Model/ForgotPasswordToken/GetCustomerByTokenTest.php

```diff
@@ -0,0 +1,145 @@
+ */
+declare(strict_types=1);
+
+namespace Magento\Customer\Test\Unit\Model\ForgotPasswordToken;
+
+use Magento\Customer\Api\CustomerRepositoryInterface;
+use Magento\Customer\Api\Data\CustomerInterface;
+use Magento\Customer\Api\Data\CustomerSearchResultsInterface;
+use Magento\Customer\Model\ForgotPasswordToken\GetCustomerByToken;
+use Magento\Framework\Api\SearchCriteria;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Exception\State\ExpiredException;
+use Magento\Framework\Phrase;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager as ObjectManagerHelper;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Test GetCustomerByToken class
+ */
+class GetCustomerByTokenTest extends TestCase
+{
+    protected const RESET_PASSWORD = 'resetPassword';
+
+    /**
+     * @var SearchCriteriaBuilder|MockObject
+     */
+    protected $searchCriteriaBuilderMock;
+
+    /**
+     * @var SearchCriteria|MockObject
+     */
+    protected $searchCriteriaMock;
+
+    /**
+     * @var CustomerRepositoryInterface|MockObject
+     */
+    protected $customerRepositoryMock;
+
+    /**
+     * @var CustomerSearchResultsInterface|MockObject
+     */
+    protected $searchResultMock;
+
+    /**
+     * @var CustomerInterface|MockObject
+     */
+    protected $customerMock;
+
+    /**
+     * @var GetCustomerByToken;
+     */
+    protected $model;
+
+    /**
+     * @inheritDoc
+     */
+    public function setUp()
+    {
+        $this->searchCriteriaBuilderMock = $this->createMock(SearchCriteriaBuilder::class);
+        $this->searchCriteriaMock = $this->createMock(SearchCriteria::class);
+        $this->searchResultMock = $this->createMock(CustomerSearchResultsInterface::class);
+        $this->customerRepositoryMock = $this->createMock(CustomerRepositoryInterface::class);
+        $this->customerMock = $this->getMockForAbstractClass(CustomerInterface::class);
+
+        $objectManagerHelper = new ObjectManagerHelper($this);
+        $this->model = $objectManagerHelper->getObject(
+            GetCustomerByToken::class,
+            [
+                'searchCriteriaBuilder' => $this->searchCriteriaBuilderMock,
+                'customerRepository' => $this->customerRepositoryMock
+            ]
+        );
+
+        $this->searchCriteriaBuilderMock->expects($this->once())
+            ->method('create')
+            ->willReturn($this->searchCriteriaMock);
+        $this->customerRepositoryMock->expects($this->once())
+            ->method('getList')
+            ->with($this->searchCriteriaMock)
+            ->willReturn($this->searchResultMock);
+    }
+
+    /**
+     * @return void
+     */
+    public function testExecute(): void
+    {
+        $totalCount = 1;
+        $this->searchResultMock->expects($this->any())
+            ->method('getTotalCount')
+            ->willReturn($totalCount);
```

Comment:
If it does not matter how many the method is being called, this means it's **stub**, not **mock**. Then you should avoid `->expects()` at all.